daemon/qrexec-daemon.c

```diff
@@ -980,16 +995,19 @@ struct QrexecPolicyRequest {
 static void send_request_to_daemon(
         const int daemon_socket,
         const char *remote_domain_name,
+        const char *source_domain,
         const char *target_domain,
         const char *service_name)
 {
     char *command;
     ssize_t bytes_sent = 0;
     int command_size = asprintf(&command,
             "source=%s\n"
+            "relayed_source=%s\n"
             "intended_target=%s\n"
             "service_and_arg=%s\n\n",
             remote_domain_name,
+            source_domain,
             target_domain,
             service_name);
```

Comment:
String manipulation in C is horrible, but you can do like this:
```suggestion
    int command_size = asprintf(&command,
            "source=%s"
            "%s%s"
            "\nintended_target=%s"
            "\nservice_and_arg=%s"
            "\n\n",
            remote_domain_name,
            source_domain ? "\nrelayed_source=" : "",
            source_domain ? : "",
            target_domain,
            service_name);
```